### PR TITLE
sql: add ImportEpoch table descriptor field

### PR DESCRIPTION
**Pull request by msbutler**
This patch introduces the new ImportEpoch table descriptor field which provides a count for the number of IMPORT INTO jobs that have been attempted on the table, including any in-progress jobs.

This work is apart of a larger project to roll back imports without MVCC timestamps. In a future pr, this epoch will get written to the imported MVCCValueHeader metadata, which in turn allows import rollbacks to use the epoch to determine which keys to delete in the table.

Informs #118280

Release note: none

### CODE REVIEW

File: pkg/sql/catalog/descpb/structured.proto [hunk 1]
```diff
@@ -1206,21 +1206,11 @@ message TableDescriptor {
...
   // ImportEpoch is the count of IMPORT INTO jobs that have been attempted on this
-  // table once it __already had data__. The ImportEpoch can include an in progress import, as the
+  // table once. The ImportEpoch can include an in progress import, as the
   // field gets incremented while preparing the table for ingestion.
   optional uint32 import_epoch = 59 [(gogoproto.nullable) = false, (gogoproto.customname) = "ImportEpoch"];
 
-  // ImportType indicates the kind of import into currently occurring on the table. 
-  enum ImportType {
-    NO_IMPORT = 0;
-    IMPORT_INTO_EMPTY = 1;
```
Comment (msbutler) [inline pkg/sql/catalog/descpb/structured.proto]: are we going to increment an import epoch for IMPORT PGDUMP? If so on restore, I think we need a way to distinguish a table apart of an IMPORT PGDUMP vs an IMPORT INTO. The former should be completely elided from the restore (the table descriptors were never online), while the latter requires a synthetic import job, right?

Comment (stevendanna) [reply pkg/sql/catalog/descpb/structured.proto]: @msbutler 

> are we going to increment an import epoch for IMPORT PGDUMP?

Per the discussion we had: The answer here is "yes for now." Import Pgdump will be excluded in much the same way it is now. We do end up giving up the fast-path rollback for existing-but-empty tables. But as that is pure optimization, we can add it later if we find we really need it.

**Comment by stevendanna** [comment]
Note to reviews: I'll be rebasing this into a single commit before merge to avoid having any commits where the tests fail.

File: pkg/sql/catalog/descpb/structured.proto [hunk 2]
```diff
@@ -1205,7 +1205,12 @@ message TableDescriptor {
   // SchemaLocked, if set, disallows schema change to this table.
   optional bool schema_locked = 58 [(gogoproto.nullable) = false, (gogoproto.customname) = "SchemaLocked"];
 
-  // Next ID: 59
+  // ImportEpoch is the count of IMPORT INTO jobs that have been attempted on this
+  // table once. The ImportEpoch can include an in progress import, as the
```
Comment (msbutler) [inline pkg/sql/catalog/descpb/structured.proto:1209]: s/once//r

**Review by msbutler — COMMENTED** [review]
I may be biased, but this looks great! GIthub will not allow to approve this pr.

**Review by dt — APPROVED** [review]

**Review by fqazi — APPROVED** [review]
:lgtm:

> Reviewed 1 of 7 files at r1, 7 of 11 files at r2, 2 of 2 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/119753#-:-NshkRaQ1KQuy7WAOHfi:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @msbutler and @rytaft)

**Comment by stevendanna** [comment]
borc cancel

**Comment by stevendanna** [comment]
bors r=dt,fqazi

**Pull request merged**